includes/Modules/Analytics_4.php

```diff
@@ -573,7 +573,7 @@ protected function create_data_request( Data_Request $data ) {
 					return new WP_Error(
 						'missing_required_setting',
 						__( 'No connected Google Analytics 4 property ID.', 'google-site-kit' ),
-						array( 'status' => 500 )
+						array( 'status' => 428 )
```

Comment:
I don't think this status is correct as it's about a [missing precondition header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/428) rather than an internal server condition.

Reply:
Ok, updated it to be `Bad Request` (`400`). The internal server (`500`) error is definitely wrong status.

Reply:
Hmm, that wouldn't be accurate either because that indicates a client error. It's quite explicit that this status means the [request shouldn't be repeated](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) as it was made which wouldn't be correct in this case because there's nothing specifically wrong with the request itself, it's an internal condition (database state) which is responsible for the error. Let's keep this as a 500 for now which doesn't feel 100% right either but is at least consistent with what we have in most other places. Then we can re-evaluate this in a follow-up issue and update it holistically once we determine what it should be 😄